### DISCUSSION TITLE
Menu and window position issue

### DISCUSSION THREAD

**Discussion by dancaer69** [labwc/labwc#953] [Q&A] [answered]
1. I have an issue with the menu in which if a submenu have many items is extended outside screen and is impossible to select any of those items. If the mouse cursor is at the bottom part of the screen when I click, then I can reach more items but this is not a solution, because still some are unreached. I've used openbox in the past but I didn't have such issue. Also there was a button if a submenu had many items to use it to scroll up/down. But on labwc there is none. Is there any fix for this?
2. Is there any way to control the position and size of a window when opens? Right now windows open on center, but I need some to open from top left side. Better if there is an option to remember the position/size as in other window managers.

**Comment by Consolatis** [comment, marked as answer]
Regarding `1.`
This is indeed a current limitation of labwc that requires some work to implement. So at the current stage I fear there is not much that can be done about it other than using more submenus to reduce the amount of items within a single menu.

Regarding `2`
> Is there any way to control the position and size of a window when opens?

You could add some manual [window rules](https://labwc.github.io/labwc-config.5.html#window_rules) like this
```xml
<windowRules>
    <windowRule identifier="some.app_id">
        <action name="MoveToEdge" direction="up" />
        <action name="MoveToEdge" direction="left" />
    </windowRule>
</windowRules>
```

> Better if there is an option to remember the position/size as in other window managers.

This feels out of scope for labwc.

**Comment by RiccardoPP** [comment]
>  2. Is there any way to control the position and size of a window when opens?

Create a new region with custom size/position and than create a **windowsRule** with a **[SnapToRegion](https://labwc.github.io/labwc-actions.5.html#entry_action_name=snaptoregion_region=value)** action.

**Reply by Consolatis** [reply]
That would be another variant, indeed. This one would provide some more flexibility as it allows defining exact position and dimensions (well, percentages of the output resolution) vs just moving the window to some edges.
